### PR TITLE
Json compare

### PR DESCRIPTION
**Pull request by bserdar**
Replaces the entity comparison algorithm with skyscreamer.JSONCompare

### CODE REVIEW

File: migrator/src/main/java/com/redhat/lightblue/migrator/Utils.java [hunk 1]
```diff
@@ -68,68 +79,109 @@ public static LightblueClient getLightblueClient(String configPath)
...
-                compareDocs(inconsistentPaths, entry.getValue(), destinationDocument.get(entry.getKey()),
-                            StringUtils.isEmpty(path) ? entry.getKey() : path + "." + entry.getKey(),excludes);
+    private static boolean matches(String path,String pattern) {
+        String[] patternComponents=pattern.split("\\.");
+        String[] pathComponents=path.split("\\.");
+        if(patternComponents.length<=pathComponents.length) {
+            for(int i=0;i<patternComponents.length;i++) {
+                if(pathComponents[i].equals(patternComponents[i])||
+                   "*".equals(patternComponents[i]))
+                    ;
```
Comment (dcrissman) [inline migrator/src/main/java/com/redhat/lightblue/migrator/Utils.java]: Looks like this if statement doesn't do anything. Should the conditional be reversed?

**Comment by dcrissman** [comment]
I would really like to see a description summarizing what this change is.

**Comment by dcrissman** [comment]
Totally as an aside, I noticed you like to use one line conditions or loops without curly braces. You do indent properly, so I'll give you that, but in general I am not a fan of this practice. I think the potential for it to be misinterpreted is not worth the handful of characters that it saves. But that is just my 2 cents.

File: migrator/src/main/java/com/redhat/lightblue/migrator/Inconsistency.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
+package com.redhat.lightblue.migrator;
+
+import java.util.List;
+
+public class Inconsistency {
```
Comment (dcrissman) [inline migrator/src/main/java/com/redhat/lightblue/migrator/Inconsistency.java]: JavaDoc?

Comment (bserdar) [inline migrator/src/main/java/com/redhat/lightblue/migrator/Inconsistency.java]: Added

**Comment by bserdar** [comment]
Added.

File: migrator/src/main/java/com/redhat/lightblue/migrator/Utils.java (hunk 1 above)
Comment (bserdar) [inline migrator/src/main/java/com/redhat/lightblue/migrator/Utils.java]: Reversed.

**Comment by dcrissman** [comment]
@bserdar looks like the build failed. Fix and I will merge.

**Pull request merged**